server/request_handler.go

```diff
@@ -71,41 +70,46 @@ func (r *RpcRequestHandler) process() {
 	defer r.req.Body.Close()
 	body, err := ioutil.ReadAll(r.req.Body)
 	if err != nil {
+		r.reqRecord.UpdateRequestEntry(r.req, http.StatusBadRequest, err.Error())
 		r.logger.Error("[process] Failed to read request body", "error", err)
 		(*r.respw).WriteHeader(http.StatusBadRequest)
 		return
 	}
 
 	if len(body) == 0 {
+		r.reqRecord.UpdateRequestEntry(r.req, http.StatusBadRequest, "empty request body")
 		(*r.respw).WriteHeader(http.StatusBadRequest)
 		return
 	}
 
 	// create rpc proxy client for making proxy request
 	client := NewRPCProxyClient(r.defaultProxyUrl)
 
+	r.reqRecord.UpdateRequestEntry(r.req, http.StatusOK, "")
 	// Parse JSON RPC payload
 	var jsonReq *types.JsonRpcRequest
 	if err = json.Unmarshal(body, &jsonReq); err != nil {
 		var jsonBatchReq []*types.JsonRpcRequest
 		if err = json.Unmarshal(body, &jsonBatchReq); err != nil {
+			r.reqRecord.UpdateRequestEntry(r.req, http.StatusBadRequest, err.Error())
 			r.logger.Error("[process] Parse payload", "error", err)
 			(*r.respw).WriteHeader(http.StatusBadRequest)
 			return
 		}
+		r.reqRecord.requestEntry.RequestType = "batch"
 		// Process batch request
 		r.processBatchRequest(client, jsonBatchReq, ip, origin, isWhitehatBundleCollection, whitehatBundleId)
 		return
 	}
-
+	r.reqRecord.requestEntry.RequestType = "single"
```

Comment:
maybe should be a bool field `isBatchRequest`, and perhaps we add `numBatchRequests` 🤔